src/shared/components/search.tsx

```diff
@@ -933,10 +930,19 @@ export class Search extends Component<any, SearchState> {
     }
   }
 
-  checkFinishedLoading() {    
-    if (this.state.searchResponse != null && this.state.resolveObjectResponse != null) {
+  checkFinishedLoading() {
+    if (
+      this.state.searchResponse != null &&
+      this.state.resolveObjectResponse != null
```

Comment:
Both are set to null when a new search is made, and set to a non-null value once the respective api response is received. In my tests it worked fine (certainly much better than the existing implementation). I think it will showa loading indicator forever if one of the requests fails with an HTTP error, as those seem to be completely unhandled. But thats also true for the existing implementation.